### ISSUE TITLE
A false negative about the rule RV_DONT_JUST_NULL_CHECK_READLINE

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1821]
Hi, I find a false negative about the rule [RV_DONT_JUST_NULL_CHECK_READLINE](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rv-method-discards-result-of-readline-after-checking-if-it-is-non-null-rv-dont-just-null-check-readline). Please refer to the below code sample.

SpotBugs should have reported a violation at line 3, but it does not. Hence, I think this is a false negative. Thanks for your consideration and I am willing to help fix this issue.  I am trying to fix this issue by modifying the [BadUseOfReturnValue.java](https://github.com/spotbugs/spotbugs/blob/51e586bed98393e53559a38c1f9bd15f54514efa/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/BadUseOfReturnValue.java) 😃 

```java
public void foo(InputStream stream, File cacheDir) throws IOException {
    FileCacheImageInputStream os = new FileCacheImageInputStream(stream, cacheDir);
    while(os.readLine() != null) {}
}
```

MySpotBugs version is 4.4.2 and I run it by Idea plugin and CLI.

**Comment by Zustin** [comment]
I have added a PR for fixing these issues #1823

**Comment by gtoison** [comment]
This was fixed in #1823
